### PR TITLE
added bookmark toggle function

### PR DESCRIPTION
**Pull request by asnatschko**

### CODE REVIEW

File: pages/_draft_story/[id].js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { shortStories } from '../stories';
+import { useContext } from 'react';
+import { BookmarksContext } from '../../components/use-bookmarks';
+
+export default function Page() {
+  const router = useRouter();
+  let id = Number(router.query.id);
+  const bookmarks = useContext(BookmarksContext);
+
+  let story = shortStories.find((x) => x.id === id);
```
Comment (ThomasKirst) [inline pages/_draft_story/[id].js:11]:
```suggestion
  const story = shortStories.find((story) => story.id === id);
```

**Review by ThomasKirst — CHANGES_REQUESTED** [review]
Hi Alex,
great you managed to add the bookmark functionality and it´s impressive that you were able to include React context here.
Personally, i would move the ´use-bookmarks.js´ file into a `hooks` folder – it´s not really a component.
I was wondering why you have the `_draft_story` folder for the detail view of a story.
All stories linked from the `stories.js` page link to a `story/[id]` page – why don´t you use it as such? Currently these links lead to a 404-page.
I left you a few minor remarks.

File: components/ButtonBookmark.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import Image from 'next/image';
+import { useContext, useState } from 'react';
+import bookmark from '../public/images/bookmark.png';
+import bookmark_filled from '../public/images/bookmark_filled.png';
+import {BookmarksContext} from '../components/use-bookmarks'
```
[the remark this answers is not in the document]
Comment (asnatschko) [reply components/ButtonBookmark.js:5]: Check:)

File: components/ButtonBookmark.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+export default function ButtonBookmark({ id }) {
+  // const [isBookmarked, setIsBookmarked] = useState();
+  const bookmarks = useContext(BookmarksContext)
+
+  const isBookmarked = bookmarks.isBookmarked(id);
+
+  function toggleBookmark() {
+    //setIsBookmarked(!isBookmarked);
```
[the remark this answers is not in the document]
Comment (asnatschko) [reply components/ButtonBookmark.js:14]: Check:)

File: components/GlobalStyle.js [hunk 4]
```diff
@@ -55,7 +55,16 @@ const GlobalStyle = createGlobalStyle`
...
+        margin-top: 20px;
+        font-size: 36px;
+        font-weight: bold;
+    }
+
+    .buttonBookmark{
+        position: relative;
+        top: 15px;
+        right: -150px;  
+        ;
```
[the remark this answers is not in the document]
Comment (asnatschko) [reply components/GlobalStyle.js:67]: Removed abandoned semikolon ))

File: components/use-bookmarks.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  const unBookmarkStory = (id) => {
+    setState([...state.filter((x) => x !== id)]);
+  };
+
+  const isBookmarked = (id) => {
+    return state.includes(id);
+  };
+
+  console.log(state);
```
[the remark this answers is not in the document]
Comment (asnatschko) [reply components/use-bookmarks.js:27]: Done :)

File: pages/stories.js [hunk 6]
```diff
@@ -22,14 +22,20 @@ const shortStories = [
...
   },
 ];
 
 export default function Stories() {
+  // const [isActive, setIsActive] = useState();
+
+  // function handleClick() {
+  //   setIsActive(!isActive);
+  // }
+
```
[the remark this answers is not in the document]
Comment (asnatschko) [reply pages/stories.js:38]: Yes :)

File: pages/_draft_story/[id].js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { useRouter } from 'next/router';
+import { shortStories } from '../stories';
+import { useContext } from 'react';
+import { BookmarksContext } from '../../components/use-bookmarks';
+
+export default function Page() {
+  const router = useRouter();
+  let id = Number(router.query.id);
```
[the remark this answers is not in the document]
Comment (asnatschko) [reply pages/_draft_story/[id].js:8]: Done :)

File: pages/_draft_story/[id].js (hunk 1 above)
Comment (asnatschko) [reply pages/_draft_story/[id].js:11]: Done :)

**Comment by asnatschko** [comment]
> Hi Alex, great you managed to add the bookmark functionality and it´s impressive that you were able to include React context here. Personally, i would move the ´use-bookmarks.js´ file into a `hooks` folder – it´s not really a component. I was wondering why you have the `_draft_story` folder for the detail view of a story. All stories linked from the `stories.js` page link to a `story/[id]` page – why don´t you use it as such? Currently these links lead to a 404-page. I left you a few minor remarks.

Hi Thomas :) I have changed your remarks and reactivated the folder draft_story :) I put it in draft, because it was not really a part of the story:) Thank you for your detailed comment!

**Review by ThomasKirst — CHANGES_REQUESTED** [review]
Well done 👍

**Pull request merged**